tests/test_sessions.py

```diff
@@ -368,11 +363,10 @@ def test_get_expired_cookies_using_max_age(self):
                 ]
             ),
             (
-                [
-                    ('Set-Cookie', 'hello=world; Path=/; Expires=Thu, 01-Jan-1970 00:00:00 GMT; HttpOnly'),
-                    ('Set-Cookie', 'pea=pod; Path=/ab; Expires=Thu, 01-Jan-1970 00:00:00 GMT; HttpOnly'),
-                    ('Connection', 'keep-alive')
-                ],
+                (
+                    'hello=world; Path=/; Expires=Thu, 01-Jan-1970 00:00:00 GMT; HttpOnly, '
+                    'pea=pod; Path=/ab; Expires=Thu, 01-Jan-1970 00:00:00 GMT; HttpOnly'
+                ),
```

Comment:
These should now be a `list`s.

Reply:
Cookies were stored as a list in `response.raw._original_response.msg._headers` however with this change cookies are extracted from `response.headers.get('Set-Cookie', '')` which is a comma separated string.

get_expired_cookies has been changed from accepting list-of-tuple headers to string of comma separated strings.

I don't see of a way to access more-or-less raw cookie data from requests in any way but parsing them out of the comma separated string without resorting to using protected members. Cookie jar has more formatted cookies but they are stored in a lossy way.

With all this in mind, to keep tests as lists the cookie splitting logic can be moved to outside of `get_expired_cookies()`, but I'm not sure if this is what you had in mind by your comment.

Thanks
--IAS

Reply:
@IlyaSukhanov Is it an option to add a test case to ensure `response.headers.get('Set-Cookie', '')` returns a string with concatenated cookies? It will help us prevent potential regressions for upcoming `requests` versions and to be more comfortable with those changes too :)

After that, we will be good to merge.

Reply:
Added test to ensure that multiple cookies get extracted correctly.

Reply:
Looks like cicd checks is failing. Is there a tried approach for starting a local http server to test against?

The way it's erroring out it's not clear what the issue is. Few of suspicions I have:

1. Multiprocess does not start for some reason
2. Multiprocess does not start fast enough (single threaded machine?)
3. Flask is unable to open port for some reason (seems unlikely as no error to that effect is presented)


Without knowing what the error is or much experience with GitHub check infrastructure. My only idea is to add a retry functionality in test, to call httpie multiple time until hopefully the background server becomes responsive.

Reply:
I was referring to the incorrect usage of tuples for what by nature is a list. 

```python
(  # <= tuple because fixed-length collection where each item is a different thing

    ( # <= list because it’s a simple list of things
        'hello=world; Path=/; Expires=Thu, 01-Jan-1970 00:00:00 GMT; HttpOnly, '
        'pea=pod; Path=/ab; Expires=Thu, 01-Jan-1970 00:00:00 GMT; HttpOnly'
    ),
    None,
    [  # <= list because it’s a simple list of things
        {'name': 'hello', 'path': '/'},
        {'name': 'pea', 'path': '/ab'}
    ]
),
```

Reply:
I think I see what you mean. But the cookie values are not tuples its a string that is line wrapped, and thus grouped using parenthesis.

Tuple:
```
(
    'foo',
    'bar'
)
```

String:
```
(
    'foo'
    'bar'
)
```

Note how in this example and in the pr the lines are not separated by comma which is required in tuple definition.

I hope this clears up any confusion.

If project has different style requirement for line wrapping long strings, please let me know and I'll make this change conform to that.

Reply:
@IlyaSukhanov right, I see it now! Yeah, this style of string-wrapping is the preferred one, but in this context, next to a bunch of tuples, it’s easy to misread it (🙋‍♂️), so I’d vote for using a different style here.